pinc/misc.inc

```diff
@@ -898,9 +914,21 @@ function is_valid_zip_file(string $file_path, bool $ignore_extension_check = fal
 
     if ($result === true) {
         $zip->close();
+    } else {
+        $zip_results = [
+            ZipArchive::ER_EXISTS => "File already exists.",
+            ZipArchive::ER_INCONS => "Zip archive inconsistent.",
+            ZipArchive::ER_INVAL => "Invalid argument.",
+            ZipArchive::ER_MEMORY => "Malloc failure.",
+            ZipArchive::ER_NOENT => "No such file.",
+            ZipArchive::ER_NOZIP => "Not a zip archive.",
+            ZipArchive::ER_OPEN => "Can't open file.",
+            ZipArchive::ER_READ => "Read error.",
+            ZipArchive::ER_SEEK => "Seek error.",
+        ];
+        $error = $zip_results[$result];
```

Comment:
do we want to handle the case some other number was in `$result`? can we add some default case to the map or should we use a switch case?

Reply:
Yes, I suppose we should allow for that case. These were the codes in the [PHP doc](https://www.php.net/manual/en/ziparchive.open.php). One of the user contributions on the page gave a longer list.